recirq/quantum_chess/mcpe_utils.py

```diff
@@ -102,14 +102,39 @@ def __init__(self, circuit: cirq.Circuit):
             for operation in moment:
                 for qubit in operation.qubits:
                     self.dependencies[qubit].append(operation)
+        self.active_gates = set()
+        for q in self.dependencies:
+            self._add_front_if_active(q)
+
+    def _add_front_if_active(self, qubit: cirq.Qid) -> None:
+        """Adds a qubit's front gate to self.active_gates if it is active."""
+        if not self.dependencies[qubit]:
+            return
+        # Get the gate at the front of the dependency list for this qubit.
+        # It's an active gate iff it is at the front of all the dependency lists
+        # of all qubits it operates on.
```

Comment:
Optional: It may be worth moving this to the docstring rather than have it be line comments.  That would help with additional visibility if people want to understand what a front gate is (like in pycharm or in recirq docs)

Reply:
Sounds good. Updated the docstring.